src/adapters/pnp/PnpAgent.c

```diff
@@ -407,7 +406,7 @@ void CloseAgent(void)
     if (NULL != g_mpiHandle)
     {
         CallMpiClose(g_mpiHandle, GetLog());
-        g_mpiHandle = NULL;
+        FREE_MEMORY(g_mpiHandle);
```

Comment:
>FREE_MEMORY(g_mpiHandle); [](http://example.com/codeflow?start=8&length=25)

The right place to free the memory for this is in the Modules Manager's MpiClose implementation, which is invoked by this code over the MPI REST API (the MPI Client -> the MPI Server -> the Modules Manager).

Reply:
I pushed the fix here: https://github.com/Azure/azure-osconfig/pull/992/commits/6c2c541180c18ffa9b762f4b5faf3addd2b97127